skfda/ml/classification/_centroid_classifiers.py

```diff
@@ -86,6 +86,9 @@ def fit(self, X: T, y: NDArrayInt) -> NearestCentroid[T]:
         Returns:
             self
         """
+        fit = getattr(self.metric, 'fit', lambda X: None)
+        fit(X)
```

Comment:
If this is going to be done several times, maybe it should be in a function in utils.